### ISSUE TITLE
The LIKE query operator is not implmented correctly.

### ISSUE THREAD

**Issue by burkenyo** [jordimontana82/fake-xrm-easy#509]
The LIKE query operator is faked to call a limited set of string methods, and it does not implement the full syntax allowed by a real organization service. In a real organization service, the LIKE operator conforms (and is likely delegated) to the underlying MS-SQL LIKE operator:

- % matches a range of characters wherever it occurs.
- _ matches a single character.
- [X-Y] matches a character within the range of characters between X and Y, inclusive.
- [XYZ] matches any of the characters X, Y, or Z.
- [^X-Y] matches a character outside the range of characters between X and Y, inclusive.
- [^XYZ] matches a character other than any of the characters X, Y, or Z.

The LIKE query operator support in FakeXrmEasy must conform to a real organization service to mock all possible queries.

**Comment by BetimBeja** [comment]
Hello @burkenyo ,
to mock all possible queries we must also handle the escaped `%` meaning `[%]`.

Best regards,
Betim Beja.

**Comment by jordimontana82** [comment]
Thanks for raising this @burkenyo

**Comment by burkenyo** [comment]
@BetimBeja, good point. Similarly `_`, `[`, and `]` can also be escaped as `[_]`, `[[]`, `[]]`.

**Comment by BetimBeja** [comment]
Hello @burkenyo ,
I submitted PR #511 which should fix this issue. If you like you can add some unit tests to my PR so we can test all the cases you mentioned.
If you are interested in Co-Authoring my Pull Request you can read my article on how to do it [here](https://www.linkedin.com/pulse/co-authoring-github-pull-requests-betim-beja/).
